sqlglot/parser.py

```diff
@@ -1969,9 +1969,32 @@ def _parse_select(
 
             self._match_r_paren()
 
-            # early return so that subquery unions aren't parsed again
-            # SELECT * FROM (SELECT 1) UNION ALL SELECT 1
-            # Union ALL should be a property of the top select node, not the subquery
+            # Parse a wrapped table as a Paren, instead of a Subquery. The exception to this
+            # is when there's an alias that can be applied to the parentheses, because that
+            # would shadow all wrapped table names, and so we want to parse it as a Subquery
+            # to represent the inner scope appropriately. Additionally, we want the node under
+            # the Subquery to be an actual query, so we will replace the table reference with
+            # a star query that selects from it.
+            if isinstance(this, (exp.Table, exp.Paren)):
+                # Removes redundant parentheses, e.g. in "SELECT * FROM ((tbl1 JOIN tbl2)) AS t"
+                this = t.cast(exp.Expression, this.unnest())
+
+                index = self._index
+                alias = self._parse_table_alias()
+
+                if not alias:
+                    return exp.paren(this, copy=False)
+
+                joins = this.args.get("joins")
+                this.set("joins", None)
+
+                this = this.replace(exp.select("*").from_(this.copy()))
+                this.set("joins", joins)
+
+                self._retreat(index)
```

Comment:
why are you retreating?

Reply:
The idea was to let the alias be parsed in `_parse_subquery` later, but seeing it again I think we can avoid retreating and attach it manually to whatever `_parse_subquery` returns.

Reply:
^ Implemented this one